BootloaderCorePkg/Stage1A/Ia32/SecEntry.nasm

```diff
@@ -62,6 +64,7 @@ FspApiSuccess:
         mov     eax, esp
 
         ; Setup HOB
+        push    ebx                  ; BistVal
```

Comment:
I don't think EBX is preserved here. Only EDI and ESI are preserved. You may need to use EDI. Can you also add a debug message in Stage1A.c to see if BistVal is expected one?

Reply:
I think we can put it back into MM0 again.  
FSP 2.0 TempRamInit() API preserves the following general purpose registers EBX, EDI, ESI, EBP and the following floating point registers MM0, MM1.

Reply:
Agree.  if we see a CPU BIST error,  we might want to print this error and then Halt the CPU.  This needs to be done after the serial port initialization.

Reply:
Thanks for the valuable points, I used MM0 register to hold the value and as it did not need to be accessed by pushing and popping in the same way as the FPU registers, I pushed the EBX register. Also, I debug the same as @myguitar suggested and confirmed with the expected value, and put a valid check and halt the CPU when BIST failure!

Reply:
Line 64 "mov     eax, esp", the "eax" is not used at all after the assignment.   So let us avoid another register "ebx" just for pushing it into stack.   I think we can safely remove line 64 and use "eax" to push into stack.

Reply:
It was actually hit me yesterday, but I was not sure why it was actually there, maybe intentionally!?

Well, I successfully boot with eax in place, implies we can remove it!